sotodlib/io/hk_utils.py

```diff
@@ -49,58 +174,67 @@ def get_grouped_hkdata(start, stop, config):
             'bf_4k' : 'observatory.LSA22HG.feeds.temperatures.Channel_06_T'
             'xy_stage_x': 'observatory.XYWing.feeds.positions.x'
             'xy_stage_y': 'observatory.XYWing.feeds.positions.y'
+
     """
     # call load_range()
     logger.debug("running load_range()")
     hkdata = load_range(start, stop, config=config)
+    _check_hkdata(hkdata)
 
     # load all fields from config file
     with open(config, 'r') as file:
         hkconfig = yaml.safe_load(file)
 
     field_dict = hkconfig['field_list']
 
-    # output a dict of field name, alias, timestamps, and data
-    # for fields that are online given start, stop and keys from load_range()
-    hk_aliases = list(hkdata.keys())
+    grouped_data = _group_data(hkdata, field_dict=field_dict, config_exists=True)
 
-    online_fields = {}
-    for alias in hk_aliases:
-        field_info = {alias: field_dict[alias]}
-        online_fields.update(field_info)
+    return grouped_data
 
-    fields_data = {}
-    for alias in online_fields:
-        time = hkdata[alias][0]
-        data = hkdata[alias][1]
-        field = online_fields[alias]
 
-        info = {field: [alias, time, data]}
-        fields_data.update(info)
+def sort_hkdata(start, stop, fields, data_dir, alias=None):
+    """
+    Takes output from load_range(), reconfigures load_range() dictionary
+    slightly. Groups HK field data by corresponding device/feed. Outputs
+    that list of data.
 
-    # now group field names by feed/device
-    sorted_fields = sorted(_get_swap_dict(online_fields))
+    Parameters:
+        start: Earliest time to search for data. Can be datetime objects,
+            unix timestamps, int, floats.
+        stop: Latest time to search for data. See start parameter above for
+            note on time formats.
+        fields: (list) List of strings of Field names to query.
+        alias (optional): (list) List of string of aliases of the field names
 
-    grouped_feeds = []
-    key_func = lambda field: field.split('.')[0:4]
-    for key, group in itertools.groupby(sorted_fields, key_func):
-        grouped_feeds.append(list(group))
+    Returns:
+        get_grouped_data (list): grouped list of data per HK feed/device;
+            useful for turning into a grouped set of axismanagers, or for
+            other grouped data analysis purposes.
 
-    # use grouped_feeds and fields_data to output a grouped list of data per
-    # feed/device, ready to input to an axismanager
-    grouped_data = []
-    for group in grouped_feeds:
-        device_data = {}
-        for field in group:
-            field_dict = {field: fields_data[field]}
-            device_data.update(field_dict)
+    """
+    if alias is None:
+        hkdata = load_range(start, stop, fields, data_dir=data_dir)
```

Comment:
when calling functions that have a lot of configurable options like `load_range`, I would always do `kwarg=kwarg`. That way if someone changes the order of the kwargs at some point these functions won't need to be changed.